src/company_cmd.cpp

```diff
@@ -603,8 +603,10 @@ static void MaybeStartNewCompany()
 	if (n < (uint)_settings_game.difficulty.max_no_competitors) {
 		/* Send a command to all clients to start up a new AI.
 		 * Works fine for Multiplayer and Singleplayer */
-		DoCommandP(0, 1 | INVALID_COMPANY << 16, 0, CMD_COMPANY_CTRL);
+		return DoCommandP(0, 1 | INVALID_COMPANY << 16, 0, CMD_COMPANY_CTRL);
```

Comment:
Beware, you generally can't use the return value of DoCommandP() as it is effectively asynchronous. If this is only on a server sending to all clients, this may be okay.

Reply:
I believe this is probably safe in this context. Any other opinions on this?

Reply:
should be safe, I can see some calls from GUI checking the return value